src/EXTRA-COMPUTE/compute_born_matrix.cpp

```diff
@@ -0,0 +1,821 @@
+      error->warning(FLERR, "Improper style does not support compute born/matrix");
+    }
+    if (kspaceflag) {
+      error->warning(FLERR, "KSPACE contribution not supported by compute born/matrix");
+    }
+  }
+
+  // need an occasional half neighbor list
+  int irequest = neighbor->request((void *) this);
+  neighbor->requests[irequest]->pair = 0;
+  neighbor->requests[irequest]->compute = 1;
+  neighbor->requests[irequest]->occasional = 1;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void ComputeBornMatrix::init_list(int /* id */, NeighList *ptr)
+{
+  list = ptr;
+}
+
+/* ----------------------------------------------------------------------
+   compute output vector
+   ------------------------------------------------------------------------- */
+
+void ComputeBornMatrix::compute_vector()
+{
+  invoked_array = update->ntimestep;
+
+  // zero out arrays for one sample
+
+  int i;
+  for (i = 0; i < nvalues; i++) values_local[i] = 0.0;
+
+  // Compute Born contribution
+  if (pairflag) compute_pairs();
+  // For now these functions are commented
+  if (bondflag) compute_bonds();
+  if (angleflag) compute_angles();
+  if (dihedflag) compute_dihedrals();
+
+  // Even if stated in Voyatzis-2012, improper and dihedrals
+  // are not exactly the same in lammps. Atoms order can depend
+  // on the forcefield/improper interaction used. As such,
+  // writing a general routine to compute improper contribution
+  // might be more tricky than expected.
+  // if (impflag) compute_impropers();
+
+  // sum Born contributions over all procs
+  MPI_Allreduce(values_local, values_global, nvalues, MPI_DOUBLE, MPI_SUM, world);
+
+  int m;
+  for (m = 0; m < nvalues; m++) { vector[m] = values_global[m]; }
+}
+
+/*------------------------------------------------------------------------
+  compute Born contribution of local proc
+  -------------------------------------------------------------------------*/
+
+void ComputeBornMatrix::compute_pairs()
+
+{
+  int i, j, m, ii, jj, inum, jnum, itype, jtype;
+  double rsq, factor_coul, factor_lj;
+  double dupair, du2pair, rinv;
+  int *ilist, *jlist, *numneigh, **firstneigh;
+
+  int *type = atom->type;
+  int *mask = atom->mask;
+  int nlocal = atom->nlocal;
+  double *special_coul = force->special_coul;
+  double *special_lj = force->special_lj;
+  int newton_pair = force->newton_pair;
+
+  // invoke half neighbor list (will copy or build if necessary)
+  neighbor->build_one(list);
+
+  inum = list->inum;
+  ilist = list->ilist;
+  numneigh = list->numneigh;
+  firstneigh = list->firstneigh;
+
+  // loop over neighbors of my atoms
+
+  Pair *pair = force->pair;
+  double **cutsq = force->pair->cutsq;
+
+  // Declares born values
+
+  int a, b, c, d;
+  double xi1, xi2, xi3;
+  double fi1, fi2, fi3;
+  double xj1, xj2, xj3;
+  double rij[3];
+  double pair_pref;
+  double r2inv;
+
+  m = 0;
+  while (m < nvalues) {
```

Comment:
It's not clear why this loop is used, since it executes only once. Is this a placeholder for a more general version of this compute?

Reply:
Hi @athomps, I used the compute_mop as a template to write this one. This loop might is from this compute which actually divided the stress in two part (kinetic and configurational).
I think that I initially wanted to do the same with the several sources of the Cij values but ended up writing several methods looping over pairs, bonds etc.

This can indeed be simplified in all the concerned methods.